### PR TITLE
improve runtime client config

### PR DESCRIPTION
**Pull request by cbellot000**
BUG: The runtime client config is only client side: it should not start a server
when a client is created a after client setting are changed: the client should take into account these changes

### CODE REVIEW

File: src/ansys/dpf/core/server_types.py [hunk 1]
```diff
@@ -1086,7 +1100,7 @@ def get_api_for_type(self, capi, grpcapi):
         return grpcapi
 
     def create_stub_if_necessary(self, stub_name, stub_type):
-        if not (stub_name in self._stubs.keys()):
+        if self.channel and not (stub_name in self._stubs.keys()):
```
Comment (cbellot000) [inline src/ansys/dpf/core/server_types.py:1103]:
```suggestion
        if self.channel and not stub_name in self._stubs:
```

File: src/ansys/dpf/core/settings.py [hunk 2]
```diff
@@ -121,8 +125,24 @@ def get_runtime_client_config(server=None):
         with Ans.Dpf.GrpcClient configuration.
 
     """
-    base = core.BaseService(server, load_operators=False)
-    return base.get_runtime_client_config()
+    from ansys.dpf.core.runtime_config import RuntimeClientConfig
+    from ansys.dpf import core
```
Comment (cbellot000) [inline src/ansys/dpf/core/settings.py:129]:
```suggestion
    from ansys.dpf import core as root
```

File: src/ansys/dpf/core/settings.py [hunk 3]
```diff
@@ -121,8 +125,24 @@ def get_runtime_client_config(server=None):
...
+    if server is None:
+        server = core.SERVER
```
Comment (cbellot000) [inline src/ansys/dpf/core/settings.py:131]:
```suggestion
        server = root.SERVER
```

File: src/ansys/dpf/gate/misc.py [hunk 4]
```diff
@@ -1,11 +1,33 @@
 DEFAULT_FILE_CHUNK_SIZE = None
 COMMON_PROGRESS_BAR = None
-_CLIENT_CONFIG = {}
+
```
Comment (PProfizi) [inline src/ansys/dpf/gate/misc.py:3]: @cbellot000 do we have an equivalent PR on pygate?

Comment (cbellot000) [reply src/ansys/dpf/gate/misc.py:3]: yes

File: src/ansys/dpf/core/core.py [hunk 5]
```diff
@@ -504,21 +504,6 @@ def release_dpf(self):
         else:
             error = self._api.data_processing_release(1)
 
-    def get_runtime_client_config(self):
```
[the remark this answers is not in the document]
Comment (cbellot000) [reply src/ansys/dpf/core/core.py:507]: yes this baseService is an internal class, users shouldn't use it

**Review by rafacanton — APPROVED** [review]

**Pull request merged**